src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -3,10 +3,12 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import React from 'react';
+import cx from 'classnames';
+import { CommonProps } from '../utils/props';
 import { useTheme } from '../utils/hooks/useTheme';
 import '@itwin/itwinui-css/css/button.css';
 
-export type ButtonGroupProps = {
+export type ButtonGroupProps = CommonProps & {
```

Comment:
This should be placed at the end, not the beginning.

Also, `title` should be excluded.

Reply:
I'm not sure why title is in CommonProps in the first place if it shouldn't be apply to any component... At this point, we should be moving style to ClassnameProps so I could use ClassnameProps instead of CommonProps.

Reply:
If you're not sure how to use our props, then you should open an issue instead of creating a PR directly.

https://github.com/iTwin/iTwinUI-react/blob/4e5c2f174e8bb2f661d04c979ffbd814e6268a13/CONTRIBUTING.md#L140

Reply:
I'm sure it doesn't make sense to have styling and classname in a separate props and having title with style in CommonProps. And still being ask to omit the title props in CommonProps when Common mean that any component could use those props... So if it's common, why should I `Omit<CommonProps, "title">` ? Why isn't style with className when both are styling related and style and title are not closely related ?

Reply:
We can look into making the `title` situation nicer but it cannot be part of this PR. Moving `style` into `ClassNameProps` is not something we can just do without checking and updating all references of `CommonProps`.

This PR is supposed to tackle a very _specific_ problem. The solution to that problem is to use `Omit<CommonProps, 'title'>`.

Reply:
CommonProps is still the same definition and there is an additive feature to ClassNamesProps which cannot cause problems, so there is no issue here... `Omit<CommonProps, "title">` is not a decent fix, it's the bandage you put on a broken bone when you should be installing a metal plate instead.

Reply:
It is kinda a breaking change... I would think about having a new name for both className and style props, if needed.
CommonProps were created for a reason. They are called common, cause the idea was to have there all the native HTML most often used attributes, which we loose support for with our custom components props (not extending the native ones). In most cases, title is used, but components as this one (some grouping ones or big ones like table), just does not make sense to have a title in there, so we need to omit it.
I would suggest just creating a new type, which could have className and style props in there.
Just not sure about the naming, but it is something connected with styling/design/etc (both of those props do the same to the element).

Reply:
Does StylingProps is reasonable ?